elements/pl-string-input/pl-string-input.py

```diff
@@ -0,0 +1,228 @@
+        with open('pl-string-input.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+
+    elif data['panel'] == 'submission':
+        parse_error = data['format_errors'].get(name, None)
+        html_params = {
+            'submission': True,
+            'label': label,
+            'parse_error': parse_error,
+            'uuid': pl.get_uuid()
+        }
+
+        if parse_error is None:
+            # Get submitted answer, raising an exception if it does not exist
+            a_sub = data['submitted_answers'].get(name, None)
+            if a_sub is None:
+                raise Exception('submitted answer is None')
+
+            # If answer is in a format generated by pl.to_json, convert it
+            # back to a standard type (otherwise, do nothing)
+            a_sub = pl.from_json(a_sub)
+
+            html_params['suffix'] = suffix
+            html_params['a_sub'] = a_sub
+        else:
+            raw_submitted_answer = data['raw_submitted_answers'].get(name, None)
+            if raw_submitted_answer is not None:
+                html_params['raw_submitted_answer'] = escape(raw_submitted_answer)
+
+        partial_score = data['partial_scores'].get(name, {'score': None})
+        score = partial_score.get('score', None)
+        if score is not None:
+            try:
+                score = float(score)
+                if score >= 1:
+                    html_params['correct'] = True
+                elif score > 0:
+                    html_params['partial'] = math.floor(score * 100)
+                else:
+                    html_params['incorrect'] = True
+            except Exception:
+                raise ValueError('invalid score' + score)
+
+        with open('pl-string-input.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+    elif data['panel'] == 'answer':
+        a_tru = pl.from_json(data['correct_answers'].get(name, None))
+        if a_tru is not None:
+            html_params = {'answer': True, 'label': label, 'a_tru': a_tru, 'suffix': suffix}
+            with open('pl-string-input.mustache', 'r', encoding='utf-8') as f:
+                html = chevron.render(f, html_params).strip()
+        else:
+            html = ''
+    else:
+        raise Exception('Invalid panel type: %s' % data['panel'])
+
+    return html
+
+
+def parse(element_html, element_index, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    name = pl.get_string_attrib(element, 'answers_name')
+    # Get allow_blank option
+    allow_blank = pl.get_string_attrib(element, 'allow_blank', False)
+
+    # Get submitted answer or return parse_error if it does not exist
+    a_sub = data['submitted_answers'].get(name, None)
+    if a_sub is None:
+        data['format_errors'][name] = 'No submitted answer.'
+        data['submitted_answers'][name] = None
+        return
+
+    try:
```

Comment:
This try/except block is more complicated than needed. I think you can just do a simple `if` statement:
```
if not a_sub and not allow_blank:
    data['format_errors'][name] = 'Invalid format. The submitted answer was left blank.'
    data['submitted_answers'][name] = None
else:
    data['submitted_answers'][name] = pl.to_json(a_sub)
```